intake_esm/cesm.py

```diff
@@ -342,51 +320,57 @@ def results(self):
+        """
+        _kwargs = self.kwargs.copy()
+        _kwargs.update(kwargs)
+        self.kwargs = _kwargs
+        return self.to_dask()
 
+    def _open_dataset(self):
+        kwargs = self._validate_kwargs(self.kwargs)
         query = dict(self.query)
-        if '*' in url or isinstance(url, list):
-            if 'concat_dim' not in kwargs.keys():
-                kwargs.update(concat_dim=self.concat_dim)
-            if self.pattern:
-                kwargs.update(preprocess=self._add_path_to_ds)
-
-            ensembles = self.query_results.ensemble.unique()
-            variables = self.query_results.variable.unique()
-
-            ds_ens_list = []
-            for ens_i in ensembles:
-                query['ensemble'] = ens_i
-
-                ds_var_list = []
-                for var_i in variables:
-
-                    query['variable'] = var_i
-                    urlpath_ei_vi = get_subset(
-                        self.collection_name,
-                        self.collection_type,
-                        query,
-                        order_by=['sequence_order', 'files'],
-                    ).files.tolist()
-
-                    dsets = [
-                        aggregate.open_dataset(url, data_vars=[var_i]) for url in urlpath_ei_vi
-                    ]
-                    ds_var_i = aggregate.concat_time_levels(dsets)
-                    ds_var_list.append(ds_var_i)
-
-                ds_ens_i = aggregate.merge(ds_var_list)
-                ds_ens_list.append(ds_ens_i)
-
-            self._ds = aggregate.concat_ensembles(ds_ens_list, member_ids=ensembles, join='outer')
-        else:
-            self._ds = xr.open_dataset(url, chunks=self.chunks, **kwargs)
+
+        ensembles = self.query_results['ensemble'].unique()
+        variables = self.query_results['variable'].unique()
+
+        ds_ens_list = []
+        for ens_i in ensembles:
+            query['ensemble'] = ens_i
+
+            ds_var_list = []
+            for var_i in variables:
+
+                query['variable'] = var_i
+                urlpath_ei_vi = get_subset(
+                    self.collection_name,
+                    self.collection_type,
+                    query,
+                    order_by=config.get('collections')['cesm']['order-by-columns'],
+                )['file_fullpath'].tolist()
+
+                dsets = [
+                    aggregate.open_dataset(
+                        url, data_vars=[var_i], decode_times=kwargs['decode_times']
+                    )
+                    for url in urlpath_ei_vi
+                ]
+                ds_var_i = aggregate.concat_time_levels(dsets, kwargs['time_coord_name'])
+                ds_var_list.append(ds_var_i)
+
+            ds_ens_i = aggregate.merge(dsets=ds_var_list)
+            ds_ens_list.append(ds_ens_i)
+
+        self._ds = aggregate.concat_ensembles(
+            ds_ens_list, member_ids=ensembles, join='outer', chunks=kwargs['chunks']
```

Comment:
I think the user needs control over `join`; we should enable this being passed in.